### PR TITLE
Padding allocates suitable buffers

### PR DESCRIPTION
**Pull request by FabioLuporini**
If padding is requested, input data is suitably mapped to a padded, temporary array, allowing the caller to completely abstract from this optimization

### CODE REVIEW

**Comment by wence-** [comment]
I'd like to land OP2/PyOP2#430, is this side of the change good to go?

File: coffee/base.py [hunk 1]
```diff
@@ -833,3 +849,26 @@ def c_flat_for(code, parent):
...
+    def __init__(self, mode):
+        self._mode = mode
+
+    def __eq__(self, other):
+        return self._mode == other._mode
+
+
+READ = Access("READ")
+WRITE = Access("WRITE")
+Rw = Access("RW")
```
Comment (kynan) [inline coffee/base.py]: Why not RW?

Comment (FabioLuporini) [inline coffee/base.py]: I just misspelled it, but there's already a commit fixing it on another branch (ast-if-nodes), which I hope will land soon

**Comment by FabioLuporini** [comment]
@wence-  Yes, this is good to go, feel free to merge it as soon as you feel ready to merge the pyop2 changes (yes, they must land together)

**Comment by wence-** [comment]
I've fixed this to DTRT in the presence of Invert nodes and rebased on master.

**Pull request merged**